sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -39,7 +39,9 @@
 #endif
 
 #ifndef __SYCL_DEPRECATED
-#ifndef SYCL_DISABLE_DEPRECATION_WARNINGS
+// The deprecated attribute is not supported in some situations(e.g. namespace)
+// in C++14 mode
+#if !defined(SYCL2020_DISABLE_DEPRECATION_WARNINGS) && __cplusplus >= 201703L
```

Comment:
We could support deprecation warnings in C++14 mode using non-standard attributes, but it would require a major refactoring since the position of C++ deprecated and non-standard attributes in declaration is different.